.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,37 @@
+name: Prepare Release
+
+# TODO: When v4 is ready to be merged this should be updated to trigger off main and support full releases via a workflow_dispatch event.
+# See https://github.com/rigetti/pyquil/issues/1501
+on:
+  push:
+    branches:
+      - v4
+  workflow_dispatch:
+    inputs:
+      type:
+        description: Bump versions and trigger a new release.
+        required: true
+        default: prerelease
+        options:
+          - prerelease
+
+jobs:
+  prepare-release:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.PAT }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        token: ${{ secrets.PAT }}
+    - name: Install Knope
+      uses: knope-dev/action@v1
+      with:
+        version: 0.6.1 # Test before updating, breaking changes likely: https://github.com/knope-dev/action#install-latest-version
+    - name: Configure Git
+      run: |
+        git config --global user.name "${{ github.triggering_actor }}"
+        git config --global user.email "${{ github.triggering_actor}}@users.noreply.github.com"
```

Comment:
Thoughts on using the name/email of github-bot rather than the triggering user? Mostly I'm unsure if the email is a valid one.

Reply:
I prefer this just for the sake of it being that much easier to see who triggered the last release, though it is a negligible difference to be sure. The email should always be valid, it's a default setup by GitHub: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-on-github